src/components/ModalDeletePieces/ModalDeletePiecesContainer.js

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useCallback, useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+
+import { stripesConnect } from '@folio/stripes/core';
+import {
+  batchFetch,
+  itemsResource,
+  locationsManifest,
+  pieceResource,
+  piecesResource,
+  requestsResource,
+  useShowCallout,
+} from '@folio/stripes-acq-components';
+
+import ModalDeletePieces from './ModalDeletePieces';
+import { getHydratedPieces } from './getHydratedPieces';
+
+const ModalDeletePiecesContainer = ({ mutator, onCancel, poLines }) => {
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const mmutator = useMemo(() => mutator, []);
+  const [pieces, setPieces] = useState();
+  const [isLoading, setIsLoading] = useState(true);
+  const showCallout = useShowCallout();
+
+  useEffect(
+    () => {
+      if (poLines?.length) {
+        batchFetch(mmutator.linePieces, poLines, (itemsChunk) => {
+          const query = itemsChunk
+            .map(({ id }) => `poLineId==${id}`)
+            .join(' or ');
+
+          return query ? `(${query}) sortby locationId` : '';
+        }, {})
+          .then(piecesResp => getHydratedPieces(piecesResp, mmutator.requests, mmutator.items, mmutator.pieceLocations))
+          .then(setPieces)
+          .then(setIsLoading);
```

Comment:
does it set result of `setPieces`? what does it return?

Reply:
could you please rephrase the question? setIsLoading doesn't set any result, it turns off the loading spinner

Reply:
changed to more explicit way